plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -391,6 +391,7 @@ public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggrega
                 return Optional.empty();
             }
 
+            // the new column can be max len(SYNTHETIC_COLUMN_NAME_PREFIX) + len(Integer.MAX_VALUE) = 9 + 10 = 19 characters which is small enough to be supported by all databases
```

Comment:
I see that explanatory description in the commit which is very elaborate. I'm just thinking whether to avoid regression here in the future, we should be adding an integration test that verifies that this 19 characters limit in case someone changes this code in the future. Instead of adding a comment, which as we know might be outdated and no tool would help us detecting that.

Reply:
Good idea. Thanks for keeping me accountable. 

I tried to think about some ways to test this but didn't spend enough time on it. I now have an idea for a test, will try it out and add it.